### ISSUE TITLE
Typekit (and such like)

### ISSUE THREAD

**Issue by ryanhyslop** [holidaypirates/nucleus#44]
Typekit doesn't seem to give you a css import option so what are peoples approaches for handling things like 3rd party embed codes?

Options I could see were:

- Overriding the theme and including it in the header (`--template` cli flag)
- Jumping on the back of [this pull request](https://github.com/holidaypirates/nucleus/pull/15) for the `@scripts` annotation and moving the embed code into a self executing function and including it in the `scripts: []` config.

ie. in nucleus config:

`"scripts": ["pattern-library/scripts/typekit-embed.js"]`

And `typekit-embed.js`:

````
(function(d) {
    var config = {
      kitId: KIT_ID,
      scriptTimeout: 3000,
      async: true
    },
    h=d.documentElement,t=setTimeout(function(){h.className=h.className.replace(/\bwf-loading\b/g,"")+" wf-inactive";},config.scriptTimeout),tk=d.createElement("script"),f=false,s=d.getElementsByTagName("script")[0],a;h.className+=" wf-loading";tk.src='https://use.typekit.net/'+config.kitId+'.js';tk.async=true;tk.onload=tk.onreadystatechange=function(){a=this.readyState;if(f||a&&a!="complete"&&a!="loaded")return;f=true;clearTimeout(t);try{Typekit.load(config)}catch(e){}};s.parentNode.insertBefore(tk,s)
  })(document);
````

The latter option is certainly preferable but this would be the first time I've had to add a specific file just for including in pattern library, and since the build step doesn't copy any of my files into the --target directory I'm currently just using browsersync to map back to that file on disk in a different directory, which doesn't seem the cleanest. 

Anyone else got some good ideas around an approach to this?

**Comment by michaelseibt** [comment]
Could you elaborate a bit what's the actual problem with Typekit? (didn't work with that yet)

I guess you need to load fonts using above script and you want them included in the styleguide?

**Comment by ryanhyslop** [comment]
Yea so Typekit specifically (and I think a few other font providers) require you to add a bit of javascript to your page for the browser to request the fonts,(https://helpx.adobe.com/typekit/using/embed-codes.html).

Some providers (such as google fonts) allow the @import rule in the stylesheet (as per the nucleus demo) which naturally works fine.

I was curious as to what other people may be doing here. Whether what I described with the scripts is a good way to go or whether other configuration options could be provided to enable 3rd party services (wonder if that would open a flood gate though).

The second option I described works fine, but raised the question (for me at least) as to how best organise scripts (or indeed other assets) that are *only* being used by the styleguide. The compiled css is also used by my app but this typekit script wont. Maybe i'm conflating this issue with two distinct issues, but was interested in seeing what approaches people had in this area.
